test/integration/dockerregistry_pullthrough_test.go

```diff
@@ -0,0 +1,313 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/distribution/manifest/schema1"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+
+	"github.com/openshift/origin/pkg/cmd/dockerregistry"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+// gzippedEmptyTar is a gzip-compressed version of an empty tar file
+// (1024 NULL bytes)
+var gzippedEmptyTar = []byte{
+	31, 139, 8, 0, 0, 9, 110, 136, 0, 255, 98, 24, 5, 163, 96, 20, 140, 88,
+	0, 8, 0, 0, 255, 255, 46, 175, 181, 239, 0, 4, 0, 0,
+}
+
+func runRegistry() error {
+	config := `version: 0.1
```

Comment:
> This is already duplicated several times.

When you say "several times" do you mean twice ? :)

No, we can't use `images/dockerregistry/config.yml` because it is not suitable for testing. In fact, even these two tests (where config placed in the string) are different. In my testcase `pullthrough: true` is required.

I can place it in `test/integration/testdata` if you want.